src/virtualdom/items/Section/prototype/setValue.js

```diff
@@ -88,15 +88,19 @@ function reevaluateSection ( section, value ) {
 		owner:      section
 	};
 
+	section.hasContext = true;
+
 	// If we already know the section type, great
 	// TODO can this be optimised? i.e. pick an reevaluateSection function during init
 	// and avoid doing this each time?
 	if ( section.subtype ) {
 		switch ( section.subtype ) {
 			case SECTION_IF:
+			section.hasContext = false;
```

Comment:
I don't think so, since `{{#foo}}` sections introduce `foo` as a context, don't they? I'm a little hazy on that, but it follows that `{{#foo}}{{bar}}{{/}}` would render `inner` with `{ foo: { bar: 'inner' }, bar: 'outer' }`. That would require the context, wouldn't it?